spectral_util/spec_io.py

```diff
@@ -131,18 +131,22 @@ def ortho_data(data, glt, glt_mask=None, glt_nodata=0, nodata_value=-9999):
     Returns:
         numpy.ndarray: The orthorectified data.
     """
+    if len(data.shape) == 2:
+        data = np.expand_dims(data, axis = -1)
     outdata = np.zeros((glt.shape[0], glt.shape[1], data.shape[2]), dtype=data.dtype) + nodata_value
     valid_glt = np.all(glt != glt_nodata, axis=-1)
     if glt_mask is not None:
         valid_glt = np.logical_and(valid_glt, glt_mask)
+    
+    glt_tmp = np.abs(glt).astype(int)
 
     if glt_nodata == 0:
-        glt[valid_glt] -= 1
-    outdata[valid_glt, :] = data[np.abs(glt[valid_glt, 1]), np.abs(glt[valid_glt, 0]), :]
+        glt_tmp[valid_glt] -= 1
+    outdata[valid_glt, :] = data[np.abs(glt_tmp[valid_glt, 1]), np.abs(glt_tmp[valid_glt, 0]), :]
+    data = np.squeeze(data) # Change it back!
```

Comment:
@jfahlen - if data is provided as a three dimensional array with one of the dimensions as one, this will pose issues.  Suggest:

- add a flag after 134 catching the case where data changes shape
- squeezing iff the flag is tripped